### PR TITLE
fix(daemon): deprecate confusing bootstrap flag

### PR DESCRIPTION
**Pull request by schomatis**
Opinionated but _backwards-compatible_ introduction of a more descriptive flag name for the old `--bootstrap` option.

### CODE REVIEW

File: cmd/lotus/daemon.go [hunk 1]
```diff
@@ -98,9 +98,24 @@ var DaemonCmd = &cli.Command{
...
 		&cli.BoolFlag{
-			Name:  "bootstrap",
-			Value: true,
+			Name:   "bootstrap",
+			Value:  true,
+			Usage:  "DEPRECATED: use 'isolated' flag instead",
+			Hidden: true,
+		},
+		&cli.BoolFlag{
+			Name:  "isolated",
```
Comment (magik6k) [inline cmd/lotus/daemon.go:116]: `--bootstrap=false` doesn't guarantee that the node won't connect to other nodes without explicitly asking it to, which `isolated`implies - if some other node connects to us we'll likely automatically connect to other nodes we get from pubsub/dht

**Review by magik6k — COMMENTED** [review]
Given that this flag is used in a rather big number of places (devnet scripts), it will probably stick around for a while.

Maybe instead of deprecating it we can just add Usage string to it which clarifies what it does without adding tech debt

File: cmd/lotus/daemon.go (hunk 1 above)
Comment (schomatis) [reply cmd/lotus/daemon.go:116]: Makes sense, what other name would be more appropriate? (Alternatively maybe we want to make it more isolated and should turn off other services besides the peer manager.)

**Comment by schomatis** [comment]
> without adding tech debt

It looks to me like the tech debt is already here the moment this flag's name is very misleading. I don't intend to change this right away but offering a better alternative (this or whatever name is more appropriate) opens the way to (slowly) repaying that debt.

If we stick with the current state just because we don't want to change the status quo we are permanently sticking with the technical debt. Again, I'm not advocating for a violent API breakage here but I think things need to change eventually and for that we need to introduce the possibility of that change in the first place.

**Pull request closed without merging**